security/access_control.rst

```diff
@@ -288,13 +305,26 @@ key:
             'access_control' => [
                 [
                     'path' => '^/_internal/secure',
-                    'allow_if' => '"127.0.0.1" == request.getClientIp() or has_role("ROLE_ADMIN")',
+                    'roles' => 'ROLE_ADMIN',
+                    'allow_if' => '"127.0.0.1" == request.getClientIp() or request.header.has('X-Secure-Access')',
                 ],
             ],
 
-In this case, when the user tries to access any URL starting with ``/_internal/secure``,
-they will only be granted access if the IP address is ``127.0.0.1`` or if
-the user has the ``ROLE_ADMIN`` role.
+In this case, when the user tries to access any URL starting with
+``/_internal/secure``, they will only be granted access if the IP address is
+``127.0.0.1`` or a secure header, or if the user has the ``ROLE_ADMIN`` role.
+
+.. note::
+
+    Internally ``allow_if`` enforce calling the built-in
+    :class:`Symfony\\Component\\Security\\Core\\Authorization\\Voter\\ExpressionVoter`
+    as like it was part of the attributes defined in the ``roles`` option.
+    By default, access is granted if one of the attribute is granted. In other
+    words, it means ``allow_if`` can grant access even if the
+    :class:`Symfony\\Component\\Security\\Core\\Authorization\\Voter\\RoleVoter` or
+    any other voter denied it.
+    To change this behavior, you may need to learn more about
+    :ref:`access decision strategy <security-voters-change-strategy>`.
```

Comment:
This whole section basically says what you've already said above - and I like the shorter description above. If you want to talk about the "access decision strategy" part - let's remove this section and add that above.